include/libdnf/rpm/package.hpp

```diff
@@ -25,12 +25,15 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #include "reldep_list.hpp"
 #include "solv_sack.hpp"
 
+#include <map>
 #include <string>
+#include <variant>
 #include <vector>
 
 
 namespace libdnf::rpm {
 
+typedef std::map< std::string, std::variant <std::string, std::vector<std::string>, Checksum, PackageId, ReldepList, bool, unsigned long, unsigned long long > > DnfVariant;
```

Comment:
As a heads up...:
- You should likely name the method something else than `serialize()`, as this is not really a serialization (I can't think of a suitable name right now though)
- A generic solution would mean overloading the `serialize()` method for all the types in question, and you can also add e.g. an overload for a vector like so (even though you may not need this, it demonstrates the concept, in case you weren't aware):
```c++
template<typename T>
std::vector<DnfVariant> serialize(const std::vector<T> & vec) {
    std::vector<DnfVariant> res;
    for (const auto & i : vec) {
        res.push_back(serialize(i));
    }
    return res;
}
```
Then you can call `serialize()` on anything you want to add to the resulting structure (assuming you've implemented `serialize(T)`).
- I'm still thinking this isn't very pretty and I also have a feeling I've run into some problems last time I attempted something like this, but I seem to have forgotten the details... Anyway, I can't think of a better solution right now.